vmdb/app/controllers/api_controller/vms.rb

```diff
@@ -92,5 +106,16 @@ def destroy_vm(vm)
     rescue => err
       action_result(false, err.to_s)
     end
+
+    def set_owner_vm(vm, owner)
+      desc = "#{vm_ident(vm)} setting owner to '#{owner}'"
+      user = User.find_or_create_by_ldap_upn(owner)
```

Comment:
The User.find_or_create_by_ldap_upn was originally created to enable VM provision requests to be sent for owners who are not yet in the db if ldap was enabled.  As the name suggests it only supports LDAP UPN configurations for external lookup.

The product has been extended to support other external systems and therefore this method would not be sufficient to support all the available use cases for creating a new user on request.

While adding a new method to support other external systems seems beyond the scope of this PR it might make more sense to impose the restriction that the user record must already exist in the db.